charts/sonarqube-dce/templates/init-sysctl.yaml

```diff
@@ -9,29 +9,41 @@ metadata:
     heritage: {{ .Release.Service }}
 data:
   init_sysctl.sh: |-
+    set -o errexit
+    set -o xtrace
     {{- if .Values.initSysctl.vmMaxMapCount }}
     if [[ "$(sysctl -n vm.max_map_count)" -lt {{ .Values.initSysctl.vmMaxMapCount }} ]]; then
       sysctl -w vm.max_map_count={{ .Values.initSysctl.vmMaxMapCount }}
+      if [[ "$(sysctl -n vm.max_map_count)" -lt {{ .Values.initSysctl.vmMaxMapCount }} ]]; then
```

Comment:
> I was thinking that we could remove the first if?

I will say no.
`initSysctl` will change the kernel value **NODE**-wide, if possible, making the change as less as possible. An example is as below:

Before the change:
```bash
joey [ ~ ]$ kubectl debug node/aks-userpool-39344234-vmss000001 -it --image=alpine -- chroot /host
Creating debugging pod node-debugger-aks-userpool-39344234-vmss000001-j9pfq with container debugger on node aks-userpool-39344234-vmss000001.
If you don't see a command prompt, try pressing enter.
root@aks-userpool-39344234-vmss000001:/# sysctl -n vm.max_map_count
65530
root@aks-userpool-39344234-vmss000001:/# exit
exit
```
Apply Pod:
```bash
joey [ ~ ]$ cat <<EOF | kubectl apply -f -
apiVersion: v1
kind: Pod
metadata:
  name: init-test
spec:
  nodeSelector:
    kubernetes.io/hostname: aks-userpool-39344234-vmss000001
  containers:
  - name: init-test
    image: ubuntu:22.04
    securityContext:
      privileged: true
    command:
    - "/bin/sh"
    - "-c"
    - |
      set -o errexit
      set -o xtrace
      sysctl -w vm.max_map_count=538157
EOF
pod/init-test created
```
After the change:
```bash
joey [ ~ ]$ kubectl debug node/aks-userpool-39344234-vmss000001 -it --image=alpine -- chroot /host
Creating debugging pod node-debugger-aks-userpool-39344234-vmss000001-mm9lb with container debugger on node aks-userpool-39344234-vmss000001.
If you don't see a command prompt, try pressing enter.
root@aks-userpool-39344234-vmss000001:/# sysctl -n vm.max_map_count
538157
```


> on a cluster that would already have the good parameter. In such case it would be needed for the user to explicitly deactivate the helm parameter.
If we don't remove the first "if", I'm sure this isn't a problem, right? And I believe if they want to run Pod on a node with good parameters which met all requirements (e.g. pre-deploy daemonset and modify all parameters needed), they should disable `initSysctl`. The scenario like this is not something we need to consider, I guess?

---
I am thinking your comment is based on "simplifying the code" as two "if" is not good-looking. But I can't think of a way to simplify the code while trying to make the change as few as possible.